src/mbgl/gl/segment.hpp

```diff
@@ -21,6 +23,21 @@ class Segment {
 
     std::size_t vertexLength;
     std::size_t indexLength;
+
+private:
+    friend class Context;
+    mutable optional<UniqueVertexArray> vao;
+};
+
+template <class Attributes>
+class SegmentVector : public std::vector<Segment> {
+public:
+    SegmentVector() = default;
+
+    template <class DrawMode>
+    SegmentVector(const VertexBuffer<typename Attributes::Vertex, DrawMode>& buffer) {
+        emplace_back(0, 0, buffer.vertexCount, 0);
+    }
```

Comment:
No, this constructor is only for unindexed rendering. I'll add a comment to clarify.